Dockerfile

```diff
@@ -23,78 +23,80 @@ RUN curl https://packages.microsoft.com/keys/microsoft.asc | apt-key add - \
     && rm -rf /var/lib/apt/lists/*
 
 ENV PATH $PATH:/opt/mssql-tools/bin
+# https://pythonspeed.com/articles/activate-virtualenv-dockerfile/
+ENV VIRTUAL_ENV=/home/default
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
 
 # From https://jdk.java.net/13/, stolen from https://github.com/docker-library/openjdk/blob/master/8/jdk/Dockerfile#L22
 ENV JAVA_HOME /usr/local/openjdk
 ENV PATH $JAVA_HOME/bin:$PATH
 RUN wget https://download.java.net/java/GA/jdk13.0.2/d4173c853231432d94f001e99d882ca7/8/GPL/openjdk-13.0.2_linux-x64_bin.tar.gz \
-	&& mkdir $JAVA_HOME \
-	&& tar xv --file openjdk-13*_bin.tar.gz --directory "$JAVA_HOME" --no-same-owner --strip-components 1 \
-	&& find "$JAVA_HOME/lib" -name '*.so' -exec dirname '{}' ';' | sort -u > /etc/ld.so.conf.d/docker-openjdk.conf \
-	&& ldconfig
+    && mkdir $JAVA_HOME \
+    && tar xv --file openjdk-13*_bin.tar.gz --directory "$JAVA_HOME" --no-same-owner --strip-components 1 \
+    && find "$JAVA_HOME/lib" -name '*.so' -exec dirname '{}' ';' | sort -u > /etc/ld.so.conf.d/docker-openjdk.conf \
+    && ldconfig
 
 # Lets always use latest pip?
 RUN /usr/local/bin/python -m pip install --upgrade pip
 
-# Install some commonly used packages and the Python application
-RUN pip3 install --use-feature=2020-resolver --no-cache-dir --upgrade --force-reinstall \
-		avro \
-		fastavro \
-		beautifulsoup4 \
-		bokeh \
-		cloudpickle \
-		colorama \
-		cython \
-		dill \
+RUN mkdir /home/default \	
+	&& python3 -m venv $VIRTUAL_ENV \	
+    && python3 -m pip install --upgrade pip \
+    && pip3 install --use-feature=2020-resolver --no-cache-dir --upgrade --force-reinstall \
+        avro \
+        fastavro \
+        beautifulsoup4 \
+        bokeh \
+        cloudpickle \
+        colorama \
+        cython \
+        dill \
         httplib2 \
         ipython \
         flake8 \
         future \
         h5py \
-        idna\<2.10\
         matplotlib \
         mlflow \
         numba \
         numexpr \
         numpy \
         nose \
-        pandas\<1.0\
+        pandas\<1.2 \
         patsy \
         pymongo \
         pyodbc \
         PyYaml \
         pytest \
         qgrid \
-        requests\<2.24.0 \

```

Comment:
I agree about removing all fixed versions, just can't remember why this was there, though most likely because of the pandas I guess